### ISSUE TITLE
Use libdeflate [de]compressor under conditional compilation

### ISSUE THREAD

**Issue by wqweto** [wqweto/ZipArchive#13]
Pros
 - better compression ratio
 - better decompression speed

Cons
 - additional dependency
 - [official release](https://github.com/ebiggers/libdeflate/releases) is a cdecl only `.dll`

**Comment by tannerhelland** [comment]
A few additional thoughts...

Pros

- better compression speed (even at minimum comp. levels)
- faster crc32

Cons

- libdeflate only implements "one-shot" deflate/inflate API.  There is no streaming API.  Probably not a deal-breaker for most users, but this would prevent future Zip64 support (via libdeflate, anyway).

Misc

- as noted in the [libdeflate README](https://github.com/ebiggers/libdeflate/blob/master/README.md), compiling via VS is "supported" but the perf hit is large.  The recommended MinGW toolchain further complicates custom stdcall building (which is why I just stick with official binaries and DispCallFunc wrappers... it's ugly, but the perf benefits are worth it IMO)
- if conditional compilation introduces too many complexities, another implementation option could be a public sub like...

```
Public Sub SetLibDeflatePath(pathToDll As String)
  mhLibDeflate = LoadLibrary(pathToDll)
  If (mhLibDeflate <> 0) Then
    'Cache proc addresses for relevant functions
  End If
End Sub
```

...with relevant deflate/inflate code gently modified...

```
If (mhLibDeflate <> 0) Then
   'Use libdeflate
Else
   'Use thunk
End If
```

Hard to know what is "best" for the average VB6 consumer of the class :/

**Comment by wqweto** [comment]
10x for input, Tanner

**Comment by tannerhelland** [comment]
Yes, [GCC does support stdcall exports](https://gcc.gnu.org/onlinedocs/gcc-4.7.2/gcc/Function-Attributes.html), but I am fuzzy on "best practices" for handling things like name-mangling with it...

tbh I have moved more and more toward p/invoke-style DispCallFunc wrapping of official 3rd-party release builds in recent years, mostly because maintaining custom stdcall builds is tedious and I find few repo owners are interested in merging patches to simplify cross-building.  But it's also entirely possible that I am just getting lazier...  :)

Either way, thank you for investigating this feature!  (And sorry for popping into this repo so often; it's one of the few I watch on GitHub as I've deployed it to quite a few client projects - across many production environments, it works excellently.)
